### DISCUSSION TITLE
Documentation translation

### DISCUSSION THREAD

**Discussion by fpoulain** [wagtail/wagtail#7691] [Ideas]
Hi,

I would like to translate Wagtail's documentation. I think starting with the editor's guide.

But afaik there is currently no translation infrastructure... Is there some hint about how to organize it?

Merry Christmas.
François

**Comment by SalahAdDin** [comment]
For Spanish too.

**Comment by gasman** [comment]
Hi - sorry for the delay in responding. We're keen to support translations of the documentation, and we're currently looking at hosting platforms that will enable this (it looks like our current documentation host, readthedocs, doesn't currently support it).

In the meantime, if you're keen to work on this, I'd suggest creating your own Github repository consisting of the files from https://github.com/wagtail/wagtail/tree/master/docs , and translating the `.rst` files one by one. We'll then be able to pull in those translations once we have the infrastructure in place.

(If I get chance, I'll try to create an example repository with the configuration in place so that you can build the docs locally.)

**Comment by children1987** [comment]
For Chinese too. Why not use transifex ?

**Comment by children1987** [comment]
BTW, for Chinese friends, will you please translate it together whit me?
https://github.com/children1987/wagtail

And it's published here:
https://wagtail-zh-hans.readthedocs.io/zh_CN/latest/editor_manual/index.html

**Comment by gasman** [comment]
@children1987 I don't think Transifex is set up to support translations of long-form text such as documentation. If it's possible at all, it will certainly need to be configured as a completely separate project - the documentation is built using Sphinx, not Django. I'm happy to consider it if someone's willing to do the research, though...

**Comment by children1987** [comment]
![image](https://user-images.githubusercontent.com/9018733/58604602-8c921f00-82c7-11e9-997d-aa224c666397.png)
I'm new about translation. 
I don't konw why you think Transifex is not fit for long-form text. As you see in the img, django-docs are also maintained in Transifex.
Would you please explain it for me? @gasman

**Comment by thibaudcolas** [comment]
It looks like things have changed since the last time there was activity on this, Read the Docs has support for documentation translations. This is documented here for the RTD parts: https://docs.readthedocs.io/en/stable/localization.html, and there is a whole guide about using Sphinx, PO files, and Transifex to manage the translation process: https://docs.readthedocs.io/en/stable/guides/manage-translations.html.

From the guide, it looks like phpmyadmin is a good example of translations done via this approach: https://github.com/phpmyadmin/localized_docs.

As @children1987 points out, Django’s docs are translated with .po files and Transifex, and I believe they also use Sphinx, but not Read the Docs https://github.com/django/django-docs-translations/tree/stable/3.0.x.

**Comment by thibaudcolas** [comment]
I have just set up French and Finnish translations for a few pages on the documentation of our Sphinx theme, https://sphinx-wagtail-theme.readthedocs.io/, as a trial:

- https://sphinx-wagtail-theme.readthedocs.io/fr/latest/
- https://sphinx-wagtail-theme.readthedocs.io/fi/latest/

This follows the RTD guides to the letter. It was pretty straightforward, there is very little configuration needed on the RTD side to get this running, and very minimal changes suggested to the Sphinx build.

If / when we start doing this for the Wagtail docs, I think we will document the process for managing documentation translations properly, and consider using Transifex for this as well. If anyone wants to take the lead on this, I think that would be very welcome. We’re pretty keen to make this happen, but this will feel much more valuable once we do have translators lining up for the content, so currently it’s a bit of a chicken-and-egg problem.

You could help by:

- Translating some of the docs (for example editor guide) in another language
- Researching and documentation translation management based on the official guides, and what Wagtail does for its UI
- Implementing fixes and improvements to the theme for translations support: https://github.com/wagtail/sphinx_wagtail_theme/issues/108

**Comment by thibaudcolas** [comment]
Discussed in today’s core team meeting – confirmed it made sense to use Transifex for this, as a separate project. Django does this as well, although in a separate repository. For Wagtail, it should be fine to keep it all in the same repository.

Based on the [documentation analytics](https://plausible.io/docs.wagtail.io/), here are popular pages to translate:

<details>

<summary>View top 20 pages</summary>

- [homepage](https://github.com/wagtail/wagtail/blob/main/docs/index.rst)
- [getting_started/tutorial](https://github.com/wagtail/wagtail/blob/main/docs/getting_started/)
- [topics/pages](https://github.com/wagtail/wagtail/blob/main/docs/topics/)
- [getting_started/index](https://github.com/wagtail/wagtail/blob/main/docs/getting_started/)
- [topics/streamfield](https://github.com/wagtail/wagtail/blob/main/docs/topics/)
- [getting_started/demo_site](https://github.com/wagtail/wagtail/blob/main/docs/getting_started/)
- [topics/writing_templates](https://github.com/wagtail/wagtail/blob/main/docs/topics/)
- [reference/pages/panels](https://github.com/wagtail/wagtail/blob/main/docs/reference/pages/)
- [topics/images](https://github.com/wagtail/wagtail/blob/main/docs/topics/)
- [topics/index](https://github.com/wagtail/wagtail/blob/main/docs/topics/)
- [reference/pages/model_reference](https://github.com/wagtail/wagtail/blob/main/docs/reference/pages/)
- [getting_started/integrating_into_django](https://github.com/wagtail/wagtail/blob/main/docs/getting_started/)
- [advanced_topics/customisation/page_editing_interface](https://github.com/wagtail/wagtail/blob/main/docs/advanced_topics/customisation/)
- [reference/hooks](https://github.com/wagtail/wagtail/blob/main/docs/reference/)
- [topics/snippets](https://github.com/wagtail/wagtail/blob/main/docs/topics/)
- [advanced_topics/i18n](https://github.com/wagtail/wagtail/blob/main/docs/advanced_topics/)
- [advanced_topics/index](https://github.com/wagtail/wagtail/blob/main/docs/advanced_topics/)
- [reference/settings](https://github.com/wagtail/wagtail/blob/main/docs/reference/)
- [getting_started/the_zen_of_wagtail](https://github.com/wagtail/wagtail/blob/main/docs/getting_started/)

</details>

Updated TODO list from the last comment (help welcome):

- [ ] Implementing fixes and improvements to the theme for translations support: https://github.com/wagtail/sphinx_wagtail_theme/issues/108
- [ ] Setting up a wagtail-docs Transifex project like Django does: https://www.transifex.com/django/django-docs/
- [ ] Researching what workflow to use to generate `.po` files for docs, upload to Transifex, and retrieve the translations.
- [ ] Documenting this workflow
- [ ] Setting up the RTD projects for languages that have complete-enough translations to be published
- [ ] Researching how to configure Algolia DocSearch for multilingual support, & making configuration changes

**Comment by thibaudcolas** [comment]
I see @children1987 had translated a lot of the editor manual to zh-Hans (simplified chinese?) a couple of years ago. I don’t think it would have changed much since so this could be a good one to use as a trial of all of the above, assuming we can find a simple way to move changes in RST files to Transifex: https://github.com/children1987/wagtail/tree/master/docs/editor_manual.

**Comment by thibaudcolas** [comment]
Converting this to GitHub Discussions, as I believe it would be a better fit for what we’re covering here.
